src/proxy/metrics.rs

```diff
@@ -153,16 +153,24 @@ impl CommonTrafficLabels {
         self
     }
 
+    fn with_derived_source_identity(mut self, w: Option<&DerivedWorkload>) -> Self {
+        let Some(w) = w else { return self };
+        // This is the identity from the TLS handshake; this is the most trustworthy source so use it
+        self.source_principal = w.identity.clone().into();
```

Comment:
should this handle if w is Some but w.identity is None? Seems like we may not want to overwrite with None.

Reply:
We actually might want to -- if the request is not over TLS I don't think we want to report the principal?

Let me check how envoy does it

Reply:
That would make sense. I'll add a hold so you can address or not but LGTM otherwise

Reply:
Envoy does NOT report the principal in metrics when doing plaintext